src/Simplify.elm

```diff
@@ -6009,6 +6032,197 @@ emptiableMapNChecks { n } emptiable checkInfo =
         Nothing
 
 
+wrapperMapNChecks : { n : Int } -> WrapperProperties otherProperties -> CheckInfo -> Maybe (Error {})
```

Comment:
I think it would be good to have some documentation (here and on similar functions) that explains what the check does. Because these have become quite generic, even by reading the code I find it pretty hard to figure out what the check does. In practice, the best way for me to find this out is by changing the code so that this function does not do anything anymore, and then look at which tests fail.

I'm thinking we could do something like this (off the top of my head, feel f

```elm
{-| Checks for `mapN` functions that where one of the wrappers causes other arguments to be discarded.

Here are the kinds of simplifications it looks for (examples using Result):

	Result.map2 f (Err y) (Ok x)
        --> Err y

        -- <and then more examples>
-}
wrapperMapNChecks : -- ...
```

What do you think? (I'm fine with adding these later and over time)

Reply:
Case in point, I'm now realizing that the simplification I added to the example is for the other check function you've introduced :sweat_smile: